Patch by MSP-Greg
May clear up some SSL verify issues, logs return length & class, URI
diff --git a/tool/downloader.rb b/tool/downloader.rb
index 13d6aa0e50..569ecc59a7 100644
--- a/tool/downloader.rb
+++ b/tool/downloader.rb
@@ -13,6 +13,7 @@
 
   # open-uri of ruby 2.2.0 accept an array of PEMs as ssl_ca_cert, but old
   # versions are not.  so, patching OpenSSL::X509::Store#add_file instead.
+if RUBY_VERSION < '2.2'
   class OpenSSL::X509::Store
     alias orig_add_file add_file
     def add_file(pems)
@@ -25,6 +26,7 @@ def add_file(pems)
       end
     end
   end
+end
   # since open-uri internally checks ssl_ca_cert using File.directory?,
   # allow to accept an array.
   class <<File
@@ -161,9 +163,11 @@ def self.download(url, name, dir = nil, since = true, options = {})
       $stdout.flush
     end
     begin
+      options[:ssl_ca_cert] = OpenSSL::X509::DEFAULT_CERT_FILE if url.scheme == 'https' && File.exist?(OpenSSL::X509::DEFAULT_CERT_FILE)
       data = with_retry(3, [Errno::ETIMEDOUT, SocketError]) do
         url.read(options.merge(http_options(file, since.nil? ? true : since)))
       end
+      puts "Downloader: data #{data.length}  class #{data.class}  #{url.to_s}"
     rescue OpenURI::HTTPError => http_error
       if http_error.message =~ /^304 / # 304 Not Modified
         if $VERBOSE
